src/dkg.rs

```diff
@@ -0,0 +1,495 @@
+        let (public_poly_c, public_proof_c, decomm_c) = polynomials_carol.public_info();
+
+        assert!(
+            polynomials_alice
+                .missing_public_polynomials()
+                .collect::<Vec<usize>>()
+                == vec![1, 2]
+        );
+
+        polynomials_alice
+            .insert_public_polynomial(1, &public_poly_b, &public_proof_b, &decomm_b)
+            .unwrap();
+        polynomials_alice
+            .insert_public_polynomial(2, &public_poly_c, &public_proof_c, &decomm_c)
+            .unwrap();
+
+        polynomials_bob
+            .insert_public_polynomial(0, &public_poly_a, &public_proof_a, &decomm_a)
+            .unwrap();
+        polynomials_bob
+            .insert_public_polynomial(2, &public_poly_c, &public_proof_c, &decomm_c)
+            .unwrap();
+
+        polynomials_carol
+            .insert_public_polynomial(0, &public_poly_a, &public_proof_a, &decomm_a)
+            .unwrap();
+        polynomials_carol
+            .insert_public_polynomial(1, &public_poly_b, &public_proof_b, &decomm_b)
+            .unwrap();
+
+        let mut dkg_alice = polynomials_alice.finish_polynomials_phase();
+        let mut dkg_bob = polynomials_bob.finish_polynomials_phase();
+        let mut dkg_carol = polynomials_carol.finish_polynomials_phase();
+
+        dkg_alice
+            .insert_secret_share(1, dkg_bob.secret_share_for_participant(0))
+            .unwrap();
+        dkg_alice
+            .insert_secret_share(2, dkg_carol.secret_share_for_participant(0))
+            .unwrap();
+
+        dkg_bob
+            .insert_secret_share(0, dkg_alice.secret_share_for_participant(1))
+            .unwrap();
+        dkg_bob
+            .insert_secret_share(2, dkg_carol.secret_share_for_participant(1))
+            .unwrap();
+
+        dkg_carol
+            .insert_secret_share(0, dkg_alice.secret_share_for_participant(2))
+            .unwrap();
+        dkg_carol
+            .insert_secret_share(1, dkg_bob.secret_share_for_participant(2))
+            .unwrap();
+
+        let alice = dkg_alice.complete().unwrap();
+        let bob = dkg_bob.complete().unwrap();
+        let _carol = dkg_carol.complete().unwrap();
+        let key_set = alice.key_set();
+
+        let ciphertext = key_set.shared_key().encrypt(15_u64, &mut rng);
+        let (alice_share, proof) = alice.decrypt_share(ciphertext, &mut rng);
+        key_set
+            .verify_share(alice_share.into(), ciphertext, 0, &proof)
+            .unwrap();
+
+        let (bob_share, proof) = bob.decrypt_share(ciphertext, &mut rng);
+        key_set
+            .verify_share(bob_share.into(), ciphertext, 1, &proof)
+            .unwrap();
+
+        let composite_dh_element =
+            *alice_share.as_element() * Scalar25519::from(2_u64) - *bob_share.as_element();
+        let message = Ristretto::mul_generator(&Scalar25519::from(15_u64));
+        assert_eq!(composite_dh_element, ciphertext.blinded_element - message);
```

Comment:
Nit: It could make sense using higher-level [`Params::combine_shares()`](https://slowli.github.io/elastic-elgamal/elastic_elgamal/sharing/struct.Params.html#method.combine_shares), as in the `sharing` module [doc example](https://slowli.github.io/elastic-elgamal/elastic_elgamal/sharing/index.html#examples).